client/src/pages/Home/index.js

```diff
@@ -1,24 +1,65 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 
 import { Pagination, ResourceList } from "../../components";
 import { useSearchParams } from "../../hooks";
 import { ResourceService, useService } from "../../services";
 
-export function Home() {
+const Home = () => {
 	const resourceService = useService(ResourceService);
 	const searchParams = useSearchParams();
-	const [{ lastPage, resources } = {}, setEnvelope] = useState();
+	const [{ lastPage, resources } = {}, setEnvelope] = useState({});
+
+	const storedPageSize = localStorage.getItem("pageSizePreference");
+	let initialPageSize;
+
+	try {
+		initialPageSize = storedPageSize ? JSON.parse(storedPageSize) : "default";
+	} catch (error) {
+		// Handle the error as needed
+		initialPageSize = "default";
+		throw new Error("Error parsing page size preference:", error);
+	}
+
+	const [pageSize, setPageSize] = useState(initialPageSize);
+
+	// This function is to save the selected page-size into local storage
+	const savePageSizePreference = (pageSize) => {
+		localStorage.setItem("pageSizePreference", JSON.stringify(pageSize));
+	};
 
 	useEffect(() => {
-		resourceService.getPublished(searchParams).then(setEnvelope);
-	}, [resourceService, searchParams]);
+		// Using the selected page size in my API request
+		resourceService
+			.getPublished({ ...searchParams, pageSize })
```

Comment:
❌ What does adding `pageSize` to the arguments here achieve? If you look at the service implementation:

https://github.com/CodeYourFuture/tech-products-demo/blob/40bf4ae5c387cc19828e1c2747f63e41fa6f5369/client/src/services/resourceService.js#L19

It destructures `page` and `perPage` out, but anything else is ignored. You could also see by looking at the Network tab in your developer tools that the request had no additional parameters. Even if that _wasn't_ true, and it passed `pageSize` along to the request, [the `GET /resources` endpoint](https://tech-products-demo-d1182e30f280.herokuapp.com/docs/#/Resources/get_resources) doesn't document a `pageSize` query parameter and would at best ignore it (search/query parameters tend to be optional, although some API implementations will validate and reject 4xx on unexpected parameters/values).